src/prettytable/prettytable.py

```diff
@@ -1590,21 +1595,23 @@ def _compute_widths(self, rows, options) -> None:
                     widths[index] = max(widths[index], self.min_width[fieldname])
         self._widths = widths
 
+        per_col_padding = sum(self._get_padding_widths(options))
         # Are we exceeding max_table_width?
         if self._max_table_width:
             table_width = self._compute_table_width(options)
             if table_width > self._max_table_width:
                 # Shrink widths in proportion
-                scale = 1.0 * self._max_table_width / table_width
-                widths = [int(w * scale) for w in widths]
+                vrulers = len(widths) - 1
```

Comment:
Can you think of another name for `vrulers`? It's not clear from the name what it's for.

Reply:
It's the number of vertical chars or how do you call `|`. What about `no_of_vchars`

Reply:
Sure, `no_of_vchars` or `num_vchars` or `n_vchars` or something?